Assets/_BForBoss/_LevelDesign/Managers/WorldManager.cs

```diff
@@ -14,16 +16,27 @@ public void CleanUp()
         
         public void Reset()
         {
-            Debug.Log("Resetting");
             _timeManager.Reset();
+            _checkpointManager.Reset();
         }
         
         private void Awake()
         {
+            _checkpointManager.Initialize();
             _stateManager.OnStateChanged += HandleStateChange;
             _stateManager.SetState(State.PreGame);
         }
         
+        //Temporary way to check if Checkpoint system works - Todo: Remove when death plane is in place
+        private void Update()
+        {
+            if (Keyboard.current[Key.R].wasPressedThisFrame)
+            {
+                GameObject player = GameObject.FindWithTag(Tags.Player);
+                player.transform.position = _checkpointManager.CurrentCheckpoint;
+            }
```

Comment:
Since there is no PlayerManager just yet, this is just a temp way of trying out the checkpoint feature